paddle/fluid/framework/fleet/fleet_wrapper.h

```diff
@@ -0,0 +1,131 @@
+/* Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#pragma once
+
+#include <memory>
+#ifdef PADDLE_WITH_PSLIB
+#include <pslib.h>
+#endif
+#include <string>
+#include <vector>
+#include "paddle/fluid/framework/scope.h"
+#include "paddle/fluid/framework/variable_helper.h"
+#include "paddle/fluid/platform/macros.h"  // for DISABLE_COPY_AND_ASSIGN
+
+namespace paddle {
+namespace framework {
+
+// A wrapper class for pslib.h, this class follows Singleton pattern
+// i.e. only initialized once in the current process
+// Example:
+//    std::shared_ptr<FleetWrapper> fleet_ptr =
+//         FleetWrapper::GetInstance();
+//    string dist_desc;
+//    fleet_ptr->InitServer(dist_desc, 0);
+// interface design principles:
+// Pull
+//   Sync: PullSparseVarsSync
+//   Async: PullSparseVarsAsync(not implemented currently)
+// Push
+//   Sync: PushSparseVarsSync
+//   Async: PushSparseVarsAsync
+// Push dense variables to server in Async mode
+// Param<in>: scope, table_id, var_names
+// Param<out>: push_sparse_status
+
+class FleetWrapper {
+ public:
+  FleetWrapper() {}
```

Comment:
Maybe we should make this default constructor a private member of `FleetWrapper`, to ensure a singleton

Reply:
done.